### PR TITLE
review Set default no classpath as true and trigger a warning when it's not set

### LINKED ISSUE #1934: Use noClasspath by default in Spoon
Most bugs we are getting there can be simply resolved by using a `setNoClasspath(true)` in the environment: why not setting this by default? 

1. It would ease the adoption of Spoon
2. For old clients we can trigger a warning to set the default value has changed
3. For old clients, as they should normally already set the classpath it should not be a problem: the classpath is still used 
4. We can always change that and release it as a major release

WDYT?

### PR DESCRIPTION
**Pull request by surli**
Fix #1934

### CODE REVIEW

File: src/main/java/spoon/support/StandardEnvironment.java [hunk 1]
```diff
@@ -458,15 +460,17 @@ public void setPreserveLineNumbers(boolean preserveLineNumbers) {
...
-
 	@Override
 	public void setNoClasspath(boolean option) {
 		noclasspath = option;
 	}
 
 	@Override
 	public boolean getNoClasspath() {
+		if (this.noclasspath == null) {
+			logger.warn("Spoon is currently use with the default noClasspath option set as true. Read the documentation for more information: http://spoon.gforge.inria.fr/launcher.html#about-the-classpath");
```
Comment (pvojtechovsky) [inline src/main/java/spoon/support/StandardEnvironment.java]: Do we want to print this warning
A) only once
B) for each call ... to fill the log by these warnings?

I vote for A) ... then add `this.noclasspath = true`

Comment (monperrus) [reply src/main/java/spoon/support/StandardEnvironment.java]: I propose to move this warning from `getNoClasspath` to a more higher level method such as `Launcher#run`

Comment (surli) [reply src/main/java/spoon/support/StandardEnvironment.java]: I'm ok with @pvojtechovsky but I don't understand the purpose of what you propose @monperrus .

**Comment by monperrus** [comment]
thanks @surli

**Pull request merged**